accounts/signals.py

```diff
@@ -0,0 +1,40 @@
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from geopy.geocoders import Nominatim
+from .models import User
+from geopy.exc import GeocoderTimedOut, GeocoderServiceError
+
+# helper to set and round coordinates
+def set_coordinates(instance, latitude, longitude):
+    # Round to 2 decimal places, make precision within 1.1km (0.7 miles)
+    instance.latitude = round(latitude, 2)
+    instance.longitude = round(longitude, 2)
```

Comment:
I noticed that this tries to round lat/long even when they’re None and caused a 500 error for bogus addresses.

I tested this change on my machine - instead of a 500 error, you get a 200 response and lat/long now show as null in Postman. I’ll share screenshots in Slack!


```suggestion
    if latitude is not None and longitude is not None:
        instance.latitude = round(latitude, 2)
        instance.longitude = round(longitude, 2)
    else:
        instance.latitude = None
        instance.longitude = None
```